File: 0234-palindrome-linked-list/0234-palindrome-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def rev(self, nd):
        prev = None
        while nd:
            after = nd.next
            nd.next = prev
            prev, nd = nd, after
        return prev
    
    def middle(self, head):
        prev = None; slow = head; fast = head
        while fast:
            prev, slow = slow, slow.next
            try:
                fast = fast.next.next
            except:
                break
        prev.next = None
        return slow
    
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        middle = self.middle(head)
        rhs = self.rev(middle)
        lhs = head

        while lhs and rhs:
            if lhs.val != rhs.val:
                return False
            lhs = lhs.next
            rhs = rhs.next
        return True

File: 0234-palindrome-linked-list/0234-palindrome-linked-list.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function toList(head: ListNode | null, acc: number[] = []) {
	if (!head) {
		return acc
	}
	acc.push(head.val)
	return toList(head.next, acc)
}

function isPalindrome(head: ListNode | null): boolean {
	const list = toList(head)
    
	//return _.isEqual(list, list.reverse() )
	return list.toString() === list.reverse().toString()
};